apis/duck/v1alpha1/addressable_types.go

```diff
@@ -26,17 +26,18 @@ import (
 
 // Addressable provides a generic mechanism for a custom resource
 // definition to indicate a destination for message delivery.
-// (Currently, only hostname, port and path are supported, and HTTP is implied. In the
-// future, additional schemes may be supported
-// ala UI may also be supported.)
-
+// Hostname is deprecated as of v0.6 and replaced with URL. It will be removed in ~v0.7
+// URL string currently supports scheme(http or https), authority(host and optional port), and path
+// Example: http://myhost:8080/ResourceGroup/ResourceName (http://<authority>/<path>)
 // Addressable is the schema for the destination information. This is
 // typically stored in the object's `status`, as this information may
 // be generated by the controller.
 type Addressable struct {
+	// Hostname is deprecated as of v0.6 and replaced with URL. It will be removed in ~v0.7
 	Hostname string `json:"hostname,omitempty"`
-	Port     string `json:"port,omitempty"`
-	Path     string `json:"path,omitempty"`
+	// URL string currently supports scheme(http or https), authority(host and optional port), and path
+	// Example: http://myhost:8080/ResourceGroup/ResourceName (http://<authority>/<path>)
+	URL string `json:"url,omitempty"`
```

Comment:
My preference is to name this `URI`, rather than `URL` to match eventing's [SubscriberSpec](https://github.com/knative/eventing/blob/d817bf6e94a278e58686cdaf219f8cb403104470/pkg/apis/eventing/v1alpha1/subscription_types.go#L152). Sadly, serving uses `URL` in [TrafficTarget](https://github.com/knative/serving/blob/538fc22770597ed95e34c5851607daf598bf0b33/pkg/apis/serving/v1alpha1/route_types.go#L88), so both have precedence.

Reply:
I would favor serving's precedent over eventing's in this case.

Reply:
I liked the URLRef idea much better and will switch to it in next iteration.

Reply:
Just to confirm, the value a user sees or specifies in a URLRef field is the URL string right? E.g.

```
status:
  address:
    url: http://myhost:8080/
```

So the use of the URLRef struct with its `Host`, `Scheme`, etc fields is an internal detail rather than something exposed to users.

Reply:
Yes.

Reply:
That is correct. I deployed it to my GKE cluster and verified it. In code it will be a struct, but what users see is the string URL in the resource yaml